src/main/java/com/kilopo/kosshop/service/Impl/ProducerServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package com.kilopo.kosshop.service.Impl;
+
+import com.kilopo.kosshop.DAO.ProducerDAO;
+import com.kilopo.kosshop.entity.Producer;
+import com.kilopo.kosshop.service.ProducerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProducerServiceImpl implements ProducerService {
+    @Autowired
+    private ProducerDAO producerDAO;
+
+    public boolean addProducer(Producer producer) {
+        if (producerDAO.addOrUpdate(producer).getName() != null) {
```

Comment:
i think better way check for id in dao class and if you addOrUpdate method return boolean you can replace it by "if (producerDAO.addOrUpdate(producer))"